src/main/java/co/rsk/federate/config/PowHSMBookkeepingConfig.java

```diff
@@ -1,87 +1,86 @@
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.maxAmountBlockHeaders")) {
-            this.maxAmountBlockHeaders = signerConfig.getConfig().getInt("bookkeeping.maxAmountBlockHeaders");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.informerInterval")) {
-            this.informerInterval = signerConfig.getConfig().getLong("bookkeeping.informerInterval");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.stopBookkeepingScheduler")) {
-            this.stopBookkeepingScheduler = signerConfig.getConfig().getBoolean("bookkeeping.stopBookkeepingScheduler");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.maxChunkSizeToHsm")) {
-            this.maxChunkSizeToHsm = signerConfig.getConfig().getInt("bookkeeping.maxChunkSizeToHsm");
+    public enum NetworkDifficultyCap {
+        MAINNET(new BigInteger("7000000000000000000000")),
+        TESTNET(BigInteger.valueOf(1000000000000000L)),
+        REGTEST(BigInteger.valueOf(20L));
+
+        private final BigInteger difficultyCap;
+
+        NetworkDifficultyCap(BigInteger difficultyCap) {
+            this.difficultyCap = difficultyCap;
         }
 
-        switch (networkParameter) {
-            case NetworkParameters.ID_MAINNET:
-                this.difficultyCap = DIFFICULTY_CAP_MAINNET;
-                break;
-            case NetworkParameters.ID_TESTNET:
-                this.difficultyCap = DIFFICULTY_CAP_TESTNET;
-                break;
-            case NetworkParameters.ID_REGTEST:
-                this.difficultyCap = DIFFICULTY_CAP_REGTEST;
-                break;
-            default:
-                String message = "Invalid network specified for the Bookkeeping Config: " + networkParameter;
-                logger.error(message);
-                throw new IllegalArgumentException(message);
+        public BigInteger getDifficultyCap() {
+            return difficultyCap;
         }
     }
 
+    private static final BigInteger DEFAULT_DIFFICULTY_TARGET = BigInteger.valueOf(3);
+    private static final boolean DEFAULT_STOP_BOOKKEPING_SCHEDULER = false;
+    private static final int DEFAULT_MAX_AMOUNT_BLOCK_HEADERS = 7;
+    private static final int DEFAULT_MAX_CHUNK_SIZE = 10;
+    private static final long DEFAULT_INFORMER_INTERVAL = 2_000;
+
+    private final SignerConfig signerConfig;
+    private final String networkParameter;
+
+    public PowHSMBookkeepingConfig(SignerConfig signerConfig, String networkParameter) {
+        this.signerConfig = signerConfig;
+        this.networkParameter = networkParameter;
+    }
+
     public BigInteger getDifficultyTarget() {
-        return difficultyTarget;
+        return signerConfig.getConfig().hasPath("bookkeeping.difficultyTarget")
+            ? new BigInteger(signerConfig.getConfig().getString("bookkeeping.difficultyTarget"))
+            : DEFAULT_DIFFICULTY_TARGET;
```

Comment:
what do you think about refactoring these methods?
To have something like

```
public BigInteger getDifficultyTarget() {

     if (!hasDifficultyTargetConfig(signerConfig)) {
          return DEFAULT_DIFFICULTY_TARGET;
     }

     return new BigInteger(signerConfig.getConfig().getString("bookkeeping.difficultyTarget"));
}

private boolean hasDifficultyTargetConfig(signerConfig) {
     return signerConfig.getConfig().hasPath("bookkeeping.difficultyTarget");
}
```

Reply:
is a good point, wdyt about this

```
public BigInteger getDifficultyTarget() {

     boolean hasDifficultyTargetConfig = signerConfig.getConfig().hasPath("bookkeeping.difficultyTarget");

     if (!hasDifficultyTargetConfig) {
          return DEFAULT_DIFFICULTY_TARGET;
     }

     return new BigInteger(signerConfig.getConfig().getString("bookkeeping.difficultyTarget"));
}
```

Reply:
IMO these snippets are just ternary operators. Unless the rest agrees, I would leave as is.